pkg/deployment/depresource/new_deployment.go

```diff
@@ -24,46 +24,45 @@ import (
 	"github.com/elastic/cloud-sdk-go/pkg/models"
 )
 
+// InstanceParams holds the common instance fields for a resource
+type InstanceParams struct {
```

Comment:
Why not follow the API naming? Maybe calling it `ResourceParams` makes more sense.

Reply:
I did at first and then it looked like `depresource.ResourceParams` a bit stuttery 😄